### PR TITLE
Add modal component to Vocabulary and Vue Vocabulary

### LINKED ISSUE #761: Component: Modal
`vocabulary` and `vue-vocabulary` both need a Modal component. This should be based off of the modal in cc search, and for `vue-vocabulary` can be copied with nearly no code changes:

https://github.com/creativecommons/cccatalog-frontend/blob/develop/src/components/AppModal.vue

Screen Shot 2020-11-04 at 12 55 55 PM

The vanilla, non-vue version should be html + css only and just display the modal. We don't need any classes to show/hide the modal, that detail can be left to the consumer of the component.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #761 by @zackkrida 

## Description

This PR adds the modal styles and the modal component from CC Search to the Vocabulary. 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: packages/vue-vocabulary/src/patterns/AppModal/AppModal.vue [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 <script>
   import { FocusTrap } from 'focus-trap-vue'
-  import Cross from '@creativecommons/fonts/src/assets/svg/symbols/times.svg?inline'
+  import Cross from '@creativecommons/fonts/assets/svg/symbols/times.svg?inline'
```
Comment (zackkrida) [inline packages/vue-vocabulary/src/patterns/AppModal/AppModal.vue:32]: Great to see the svgs in action 😄

**Review by zackkrida — APPROVED** [review]
Great work here too! My only comment is—any ideas on how we can fix the rendering of the modal in the storybook `Docs` view?

Screen Shot 2020-11-12 at 11 19 21 AM

Maybe in the future we'll just split the `AppModal` up into subcomponents, and then only show the inner sub-component in the Docs view.

**Comment by obulat** [comment]
>any ideas on how we can fix the rendering of the modal in the storybook `Docs` view?
> 
Do you mean how we can make the modal appear/disappear on a button click, or something similar? I couldn't figure out how to add functionality to HTML storybook ...

**Comment by zackkrida** [comment]
> > any ideas on how we can fix the rendering of the modal in the storybook `Docs` view?
> 
> Do you mean how we can make the modal appear/disappear on a button click, or something similar? I couldn't figure out how to add functionality to HTML storybook ...

No I mean how the modal fills the "Docs" tab when it would be preferable to have it contained in the preview area (the grey box with rounded corners and the "code" tab).

**Pull request merged**